pyproject.toml

```diff
@@ -22,10 +22,23 @@ classifiers = [
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-    "jupyter_server>=2.0.1,<3"
+    "jupyter_server>=2.0.1,<3",
+    "pydantic>=2.0"
```

Comment:
It looks like this is used only for `NotificationParams`. Could it be a `dataclass` instead? This is because `pydantic` in past broke many projects with major release causing a lot of maintenance headaches, so it if it is only used for a single class then maybe it is not worth it.